aws_lambda_powertools/utilities/data_classes/s3_batch_operation_event.py

```diff
@@ -0,0 +1,237 @@
+    """S3 Batch Operations response object
+
+    Documentation:
+    --------------
+    - https://docs.aws.amazon.com/lambda/latest/dg/services-s3-batch.html
+    - https://docs.aws.amazon.com/AmazonS3/latest/userguide/batch-ops-invoke-lambda.html#batch-ops-invoke-lambda-custom-functions
+    - https://docs.aws.amazon.com/AmazonS3/latest/API/API_control_LambdaInvokeOperation.html#AmazonS3-Type-control_LambdaInvokeOperation-InvocationSchemaVersion
+
+    Parameters
+    ----------
+    invocation_schema_version : str
+        Specifies the schema version for the payload that Batch Operations sends when invoking
+        an AWS Lambda function., either '1.0' or '2.0'. This must be copied from the event.
+
+    invocation_id : str
+        The identifier of the invocation request. This must be copied from the event.
+
+    treat_missing_keys_as : Literal["Succeeded", "TemporaryFailure", "PermanentFailure"]
+        undocumented parameter, defaults to "PermanentFailure"
+
+    results : List[S3BatchOperationResult]
+        results of each S3 Batch Operations task,
+        optional parameter at start. can be added later using `add_result` function.
+
+    Examples
+    --------
+
+    **S3 Batch Operations**
+
+    ```python
+        import boto3
+
+        from botocore.exceptions import ClientError
+
+        from aws_lambda_powertools.utilities.data_classes import (
+            S3BatchOperationEvent,
+            S3BatchOperationResponse,
+            event_source
+        )
+        from aws_lambda_powertools.utilities.typing import LambdaContext
+
+
+        @event_source(data_class=S3BatchOperationEvent)
+        def lambda_handler(event: S3BatchOperationEvent, context: LambdaContext):
+            response = S3BatchOperationResponse(event.invocation_schema_version, event.invocation_id, "PermanentFailure")
+
+            result = None
+            task = event.task
+            src_key: str = task.s3_key
+            src_bucket: str = task.s3_bucket
+
+            s3 = boto3.client("s3", region_name='us-east-1')
+
+            try:
+                dest_bucket, dest_key = do_some_work(s3, src_bucket, src_key)
+                result = task.build_task_batch_response("Succeeded", f"s3://{dest_bucket}/{dest_key}")
+            except ClientError as e:
+                error_code = e.response['Error']['Code']
+                error_message = e.response['Error']['Message']
+                if error_code == 'RequestTimeout':
+                    result = task.build_task_batch_response("TemporaryFailure", "Retry request to Amazon S3 due to timeout.")
+                else:
+                    result = task.build_task_batch_response("PermanentFailure", f"{error_code}: {error_message}")
+            except Exception as e:
+                result = task.build_task_batch_response("PermanentFailure", str(e))
+            finally:
+                response.add_result(result)
+
+            return response.asdict()
+
+    ```
+    """  # noqa: E501
```

Comment:
Consider actually solving the warning. Too long lines will not look good when the IDE shows inline documentation.